vhost-user-backend/Cargo.toml

```diff
@@ -11,6 +11,7 @@ license = "Apache-2.0"
 [features]
 xen = ["vm-memory/xen", "vhost/xen"]
 postcopy = ["vhost/postcopy", "userfaultfd"]
+gpu-set-socket = ["vhost/vhost-user-gpu-set-socket"]
```

Comment:
this feature was added for the `gpu-set-socket` functionality. Given that in QEMU for vhost-user-gpu the expectation is different compared to other backend, it requires a VHOST_USER_GPU_SET_SOCKET to obtain the socket. Also this helps to manage the dependencies for the vost-device-gpu crate, since other crates don't need it.

Reply:
What dependencies depend on this feature?

If we save little code and no dependencies, I don't know how worthwhile it is.

Reply:
The `gpu-set-socket` is here to enable the  `vhost-user-gpu-set-socket` feature in the `vhost` dependency. There aren't any external dependencies this brings in. I put it behind a feature flag, because this feature is useless for every other device apart from a gpu device.

I can remove this flag and have it be always enabled if you want. If we decide to keep it I'll make sure to add it to CI.

Reply:
> The `gpu-set-socket` is here to enable the `vhost-user-gpu-set-socket` feature in the `vhost` dependency. There aren't any external dependencies this brings in. I put it behind a feature flag, because this feature is useless for every other device apart from a gpu device.

Yep, I meant both features. Sorry to be unclear.

> 
> I can remove this flag and have it be always enabled if you want. If we decide to keep it I'll make sure to add it to CI.

I'm in doubt, on the one hand I understand why, on the other hand we don't want to add too many features. But maybe in this case it makes sense, because a generic backend doesn't want to handle those types of messages at all.

If we want to keep it, what about a shorter name like `gpu` ?

Reply:
Sure, how about naming the feature `gpu-socket` in both `vhost` and `vhost-user-backend` ? This seem a bit more clear than `gpu` alone.

Reply:
> Sure, how about naming the feature `gpu-socket` in both `vhost` and `vhost-user-backend` ? This seem a bit more clear than `gpu` alone.

yes, I was going to write "or gpu-socket" in the previous message, so if you also think that is okay, I agree to use `gpu-socket` for both ;-)